sgkit/stats/regenie.py

```diff
@@ -0,0 +1,778 @@
+from typing import Any, Dict, Hashable, Optional, Sequence, Tuple, Union
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from dask.array import Array
+from numpy import ndarray
+from xarray import Dataset
+
+from ..typing import ArrayLike
+from ..utils import split_array_chunks
+from .utils import (
+    assert_array_shape,
+    assert_block_shape,
+    assert_chunk_shape,
+    concat_2d,
+    r2_score,
+)
+
+
+def index_array_blocks(
+    x: Union[ArrayLike, Sequence[int]], size: int
+) -> Tuple[ndarray, ndarray]:
```

Comment:
Should this be `Tuple[ArrayLike, ArrayLike]`?

Reply:
This one is maybe slightly different from `index_block_sizes` in that I suppose a provided dask array could remain a dask array.  I'd rather not then have to test that scenario because I can't imagine why we'd need it.